src/i18n.ts

```diff
@@ -0,0 +1,75 @@
+import i18n from "i18next";
+import { getRuntime } from "@yext/pages/util";
+import {
+  useTranslation as usei18nextTranslation,
+  initReactI18next,
+} from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
+import type { TemplateRenderProps } from "src/types/entities";
+
+declare global {
+  interface Window {
+    __INITIAL__DATA__: TemplateRenderProps;
+  }
+}
+
+const detector = {
+  name: "pagesJSDetector",
+
+  lookup() {
+    if (getRuntime().name === "browser") {
+      return window.__INITIAL__DATA__.document.locale as string;
+    }
+  },
+
+  cacheUserLanguage() {
+    // Intentionally unimplemented
+  },
+};
+
+const languageDetector = new LanguageDetector();
+languageDetector.addDetector(detector);
+
+const dev = import.meta.env.DEV;
+const lookupOrder = dev ? ["querystring"] : ["pagesJSDetector"];
```

Comment:
When running locally we look for a `locale` query string, which is the same thing the pagesjs dev server uses. In production we check the locale of the document through `__INITIAL__DATA__`.

Reply:
It sounds like this applies translations client-side, but we need the pages to be translated when they're generated. We have at least one client, Accor, that will not accept client-side translations. It also hurts page speed and may cause crawlers that don't run JS to get untranslated content.

EDIT: After getting farther along in this PR, I see the `useSSR` hook. Does that enable translations during the page generation?

Reply:
I guess I assumed that it was doing that, but I see that it isn't. Will look into it, I agree about this being a hard requirement.

Reply:
This should be working now. If you open this page for example with javascript disabled you should see the translated content: https://i18nextusage-saucily--lavender--seahorse-pgsdemo-com.preview.pagescdn.com/es/ny/new-york/61-ninth-avenue